src/app/lib/globalStyles/index.jsx

```diff
@@ -22,9 +23,17 @@ const GlobalStyleComponent = createGlobalStyle`
     ${({ fonts }) => fonts && createFontStyles(fonts)}
   `;
 
-const GlobalStyle = () => {
-  const { fonts } = useContext(ServiceContext);
-  return <GlobalStyleComponent fonts={fonts} />;
+const GlobalStyle = ({ fonts }) => {
+  const styleFonts = useContext(ServiceContext).fonts;
```

Comment:
Can we not do 
```jsx
const styleFonts = fonts || useContext(ServiceContext).fonts;
``` 
to avoid calling useContext unnecessarily?

Reply:
@dr3 do you mean `can we do` the above suggested implementation? am lost at `not`

Reply:
with the above suggestion, it will violate the rules of hooks as it’s not advisable to use conditionals with hooks.

Reply:
Ooooh yes good point. Ignore me :)